source/common/upstream/upstream_impl.cc

```diff
@@ -2141,19 +2075,9 @@ void PriorityStateManager::updateClusterPrioritySet(
   LocalityWeightsSharedPtr locality_weights;
   std::vector<HostVector> per_locality;
 
-  // If we are configured for locality weighted LB we populate the locality weights. We also
-  // populate locality weights if the cluster uses load balancing extensions, since the extension
-  // may want to make use of locality weights and we cannot tell by inspecting the config whether
-  // this is the case.
-  //
   // TODO: have the load balancing extension indicate, programmatically, whether it needs locality
   // weights, as an optimization in cases where it doesn't.
-  const bool locality_weighted_lb =
-      parent_.info()->lbConfig().has_locality_weighted_lb_config() ||
-      parent_.info()->lbType() == LoadBalancerType::LoadBalancingPolicyConfig;
-  if (locality_weighted_lb) {
-    locality_weights = std::make_shared<LocalityWeights>();
-  }
+  locality_weights = std::make_shared<LocalityWeights>();
```

Comment:
why always create this now?

Reply:
See, because the `locality_weighted_lb` always be true now.

Reply:
why?  why doesn't that depend on what lb algorithm we use?
/wait-any

Reply:
This is long time historical reason. (before I start to take this part of code). See the comments and the TODO:
```
  // We also
  // populate locality weights if the cluster uses load balancing extensions, since the extension
  // may want to make use of locality weights and we cannot tell by inspecting the config whether
  // this is the case.

  // TODO: have the load balancing extension indicate, programmatically, whether it needs locality
  // weights, as an optimization in cases where it doesn't.
```

This PR acutally doesn't change anything. Because the `locality_weighted_lb` will be true and the locality_weights will be created anyway after we convert the legacy lb config to the extension lb config.